tests/sync/app.cpp

```diff
@@ -125,12 +170,217 @@ TEST_CASE("app: login_with_credentials integration", "[sync][app]") {
     }
 }
 
+// MARK: - UsernamePasswordProviderClient Tests
+
+TEST_CASE("app: UsernamePasswordProviderClient integration", "[sync][app]") {
+
+    auto email = util::format("%1@%2.com", random_string(10), random_string(10));
+    auto password = random_string(10);
+    
+    std::unique_ptr<GenericNetworkTransport> (*factory)() = []{
+        return std::unique_ptr<GenericNetworkTransport>(new IntTestTransport);
+    };
+    
+    auto config = App::Config{"translate-utwuv", factory};
+    auto app = App(config);
+    try_make_dir("./" + config.app_id);
+    SyncManager::shared().configure({ .base_file_path = "./" + config.app_id });
```

Comment:
Tests need to use `TestSyncManager` rather than calling `configure()` directly. Calling configure() like this resets everything but the base path to the default (resulting in an invalid config on linux), and fails to clean up the state at the end of the test.

```
    std::string base_path = tmp_dir() + "/" + config.app_id;
    reset_test_directory(base_path);
    TestSyncManager init_sync_manager(base_path);
```

Reply:
Noted, I have moved to using `TestSyncManager ` in the tests